### PR TITLE
VUE-188 KvPillToggle Storybook and a11y improvements

### PR DESCRIPTION
**Pull request by ryan-ludwig**
- Change from @ clicks on spans to radio buttons for a11y. Add disabled, hover, active, and focus styles
- Add .stories for KvPillToggle
- Create Form Elements subfolder in Storybook

No visual differences from the previous component aside from the hover, focus, and active states being added. No changes to the props or events. Now usable with a keyboard. 

Verified on 
https://www.kiva.org/checkout
https://www.kiva.org/lend/filter?gender=female&sortBy=popularity

### CODE REVIEW

File: src/components/Checkout/BraintreeCheckout.vue [hunk 1]
```diff
@@ -104,11 +104,11 @@
 					<div id="kv-expiration-date" class="kv-braintree-wrapper"></div>
 				</div>
 				<div class="small-4 columns">
```
Comment (ryan-ludwig) [inline src/components/Checkout/BraintreeCheckout.vue]: Sneaking this in, not related to KvPillToggle, just noticed it on the same page.

File: src/components/Kv/KvPillToggle.vue [hunk 2]
```diff
@@ -1,34 +1,62 @@
...
+		<div
+			v-for="option in options"
+			:key="option.key"
+			class="pill"
 		>
-			<div class="title">
+			<input
+				class="radio"
+				type="radio"
+				:name="`pill-${uuid}`"
```
Comment (mcstover) [inline src/components/Kv/KvPillToggle.vue]: why the uuid structure instead of just pulling index off of the v-for?

Comment (ryan-ludwig) [reply src/components/Kv/KvPillToggle.vue]: It showed up when I put multiple kv-pill-toggle elements on a page that they'll have the same names / ids. So you click an input on the bottom kv-pill-toggle and the top one updates!

Comment (mcstover) [reply src/components/Kv/KvPillToggle.vue]: Ahhh ok... So this is really a form input/label correlation specific issue. Just want to be careful of propagating this type of extra code throughout other components.

Should be ok with regard to form elements as there are only so many of them, though if it starts to go beyond that let's find a better way to eliminate the boilerplate.

Comment (eddieferrer) [reply src/components/Kv/KvPillToggle.vue]: I don't like this. Is this a bug with storyBook? multiple components on the same page won't do this with Vue. If it is storybook related the fix should be in the story, not in the component

Comment (ryan-ludwig) [reply src/components/Kv/KvPillToggle.vue]: @eddieferrer You're right! The bug wasn't in Storybook bug though, it was that my demo had all of the instances bound to the same Vue model. Doh! Will push a fix soon.

Comment (ryan-ludwig) [reply src/components/Kv/KvPillToggle.vue]: Well now it seems like it was a combination of both duplicate model bindings and also duplicate IDs. By removing the IDs and for attributes on the labels and wrapping them instead it seems we're OK. For reference, here's a [paired down example](https://codesandbox.io/s/relaxed-cherry-1qfo2) of 
`<input id="test"><label for="test">`
 vs 
`<label><input></label>`.

File: src/components/Kv/KvPillToggle.vue [hunk 3]
```diff
@@ -17,17 +26,34 @@ export default {
...
 		},
 		selected: {
 			type: String,
-			required: true
+			default: '',
 		},
 	},
+	data() {
+		return {
+			checked: ''
```
Comment (eddieferrer) [inline src/components/Kv/KvPillToggle.vue]: Alternatively you can just bind checked to selected here and avoid the created hook all together...
```suggestion
			checked: this.selected
```

Comment (ryan-ludwig) [reply src/components/Kv/KvPillToggle.vue]: Nice, thanks!

**Pull request merged**